src/interface/mapping/iter.rs

```diff
@@ -0,0 +1,210 @@
+                Self::Individual(individual.iter_mappings())
+            }
+            InterfaceIml::DatastreamObject(object) => Self::Object(object.iter_mappings()),
+            InterfaceIml::Properties(properties) => Self::Properties(properties.iter_mappings()),
+        }
+    }
+}
+
+impl<'a> Iterator for MappingIter<'a> {
+    type Item = Mapping<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self {
+            Self::Properties(iter) => iter.next(),
+            Self::Individual(iter) => iter.next(),
+            Self::Object(iter) => iter.next(),
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        match self {
+            Self::Properties(iter) => iter.size_hint(),
+            Self::Individual(iter) => iter.size_hint(),
+            Self::Object(iter) => iter.size_hint(),
+        }
+    }
+}
+
+impl DoubleEndedIterator for MappingIter<'_> {
+    fn next_back(&mut self) -> Option<Self::Item> {
+        match self {
+            Self::Properties(iter) => iter.next_back(),
+            Self::Individual(iter) => iter.next_back(),
+            Self::Object(iter) => iter.next_back(),
+        }
+    }
+}
+
+impl ExactSizeIterator for MappingIter<'_> {
+    fn len(&self) -> usize {
+        match self {
+            Self::Properties(iter) => iter.len(),
+            Self::Individual(iter) => iter.len(),
+            Self::Object(iter) => iter.len(),
+        }
+    }
+}
+
+impl FusedIterator for MappingIter<'_> {}
+
+#[derive(Debug, Clone)]
+pub(crate) struct PropertiesMappingIter<'a> {
+    properties: ValuesIter<'a, PropertiesMapping>,
+}
+
+impl<'a> PropertiesMappingIter<'a> {
+    pub(crate) fn new(properties: &'a MappingMap<PropertiesMapping>) -> Self {
+        Self {
+            properties: properties.values(),
+        }
+    }
+}
+
+impl<'a> Iterator for PropertiesMappingIter<'a> {
+    type Item = Mapping<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.properties.next().map(|mapping| mapping.into())
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.properties.size_hint()
+    }
+}
+
+impl DoubleEndedIterator for PropertiesMappingIter<'_> {
+    fn next_back(&mut self) -> Option<Self::Item> {
+        self.properties.next_back().map(|mapping| mapping.into())
+    }
+}
+
+impl ExactSizeIterator for PropertiesMappingIter<'_> {
+    fn len(&self) -> usize {
+        self.properties.len()
+    }
+}
+
+impl FusedIterator for PropertiesMappingIter<'_> {}
+
+#[derive(Debug, Clone)]
+pub(crate) struct IndividualMappingIter<'a> {
+    properties: ValuesIter<'a, DatastreamIndividualMapping>,
+}
+
+impl<'a> IndividualMappingIter<'a> {
+    pub(crate) fn new(properties: &'a MappingMap<DatastreamIndividualMapping>) -> Self {
+        Self {
+            properties: properties.values(),
+        }
+    }
+}
+
+impl<'a> Iterator for IndividualMappingIter<'a> {
+    type Item = Mapping<'a>;
```

Comment:
This is not possible, since the iterator is used to make the interfaces uniform. Mapping lets us have the same properties and information for all the types.

It's a single type to iniform for all the different interfaces.